Updating Trident on ARO article based on ARO 4.20 validation (#880)
rh-mobb/documentation: content/aro/trident/index.md

---
date: '2022-05-23'
title: Trident operator setup for Azure NetApp Files on ARO
tags: ["ARO"]
authors:
  - Byron Miller
  - Connor Wooley
  - Kevin Collins
  - Diana Sari
validated_version: "4.20"
---

Note:

>  This guide is a simple "happy path" that shows a minimal friction way to use Azure NetApp Files with Azure Red Hat OpenShift. This may not be the best behavior for any system beyond demonstration purposes.

Reviewer: Convert plain `Note:` text to RHDS alert shortcode per AGENTS.md.

```suggestion
{{< alert state="info" >}}
This guide is a simple "happy path" that shows a minimal friction way to use Azure NetApp Files with Azure Red Hat OpenShift. This may not be the best behavior for any system beyond demonstration purposes.
{{< /alert >}}
```

### Delegate subnet to ARO

Login to the Azure console, find the VNet used by your ARO cluster, and add a delegated subnet for Azure NetApp Files. Make sure the backend configuration later in this guide references the exact subnet name/path you created.

## Install Trident Operator from OperatorHub/Software Catalog

Login to your ARO cluster and install **NetApp Trident** from **OperatorHub** (or **Software Catalog**) using the certified operator.

1. In the OpenShift console, go to **OperatorHub**.
2. Search for **NetApp Trident**.
3. Select the most recent available operator version.
4. Install the operator in the default recommended configuration.
5. Create a `TridentOrchestrator` instance.

Example:

```yaml
apiVersion: trident.netapp.io/v1
kind: TridentOrchestrator
metadata:
  name: trident

>   namespace: openshift-operatorsß

Reviewer: Fix stray `ß` character — this renders verbatim in the live preview and produces invalid YAML if copy-pasted.

```suggestion
  namespace: openshift-operators
```

spec:
  namespace: trident
  IPv6: false
```

Apply and verify:

```bash
oc apply -f tridentorchestrator.yaml
oc get tridentorchestrator -n openshift-operators
oc get pods -n trident
```

## Create Trident backend

Create the backend using a Kubernetes Secret and a `TridentBackendConfig` custom resource.

Create the credentials secret first:

```bash
oc -n trident create secret generic anf-credentials \
  --from-literal=clientID="<app-id>" \
  --from-literal=clientSecret="<app-secret>"
```

> Notes:
>
> * Ensure the service principal has the required Azure permissions for Azure NetApp Files resources.

> > * If permissions are missing, you may see an error similar to: `capacity pool query returned no data; no capacity pools found for storage pool`.

Reviewer: Convert raw blockquote to RHDS alert shortcode.

```suggestion
{{< alert state="info" >}}
* Ensure the service principal has the required Azure permissions for Azure NetApp Files resources.
* If permissions are missing, you may see an error similar to: `capacity pool query returned no data; no capacity pools found for storage pool`.
{{< /alert >}}
```

```bash
NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
anf-pvc   Bound    pvc-b835e6c6-3f3e-4a6b-aeb3-3b2906df3bd5   100Gi      RWX            anf-sc         <unset>                 15s
```

Check the PersistentVolumes:

```bash
oc get pv
```

Example output:

```bash
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM              STORAGECLASS   VOLUMEATTRIBUTESCLASS   REASON   AGE
pvc-b835e6c6-3f3e-4a6b-aeb3-3b2906df3bd5   100Gi      RWX            Delete           Bound    netappdemo/anf-pvc anf-sc         <unset>                          15s
```

### Verify in OpenShift Console

Login to the cluster as `cluster-admin` and confirm that:

* the `anf-sc` StorageClass is present
* the `anf-pvc` claim in the `netappdemo` project is `Bound`
* a dynamically provisioned PersistentVolume was created for the claim

## Create Pods to test Azure NetApp

Create two pods to validate the Azure NetApp file mount. One pod writes data to the shared volume, and the second pod reads the same data back to confirm `ReadWriteMany` access is working correctly.

> > **Note:** On current OpenShift clusters, these simple demo pods may emit Pod Security warnings unless restricted-compatible `securityContext` settings are added. The sample still works for basic validation.

Reviewer: Convert raw blockquote to RHDS alert shortcode.

```suggestion
{{< alert state="info" >}}
On current OpenShift clusters, these simple demo pods may emit Pod Security warnings unless restricted-compatible `securityContext` settings are added. The sample still works for basic validation.
{{< /alert >}}
```
